make_a_plea/management/commands/zendesk.py

```diff
@@ -0,0 +1,234 @@
+                    self.writerow(self.ticket_to_row(ticket))
+
+    def writerow(self, row):
+        self.writer.writerow([s.encode("utf-8") for s in row])
+        data = self.queue.getvalue()
+        data = data.decode("utf-8")
+        data = self.encoder.encode(data)
+        self.stream.write(data)
+        self.queue.truncate(0)
+
+    def ticket_to_row(self, ticket):
+        return \
+            self.get_simple_fields(ticket) + \
+            self.get_newline_fields(ticket) + \
+            self.get_structured_fields(ticket)
+
+    def parse_simple_field(self, field):
+        try:
+            return field.encode("utf-8")
+        except AttributeError:  # ints
+            return str(field).encode("utf-8")
+
+    def get_simple_fields(self, ticket):
+        fields = []
+        for field in SIMPLE_FIELDS:
+            if field in ticket:
+                fields.append(self.parse_simple_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
+
+    def parse_newline_field(self, field):
+        """Newlines make for a messy csv so coerce them to \n"""
+        rep = {
+            "\\u000A".decode('unicode-escape'): "/n",
+            r"/r": "/n",
+            r"/r/n": "/n",
+            r"/n": "/n",
+        }
+        rep = dict((re.escape(k), v) for k, v in rep.iteritems())
+        pattern = re.compile("|".join(rep.keys()))
+        try:
+            return pattern.sub(
+                lambda m: rep[re.escape(m.group(0))], field, re.UNICODE)
+        except TypeError:
+            return ""
+
+    def get_newline_fields(self, ticket):
+        fields = []
+        for field in NEWLINE_FIELDS:
+            if field in ticket and field is not None:
+                fields.append(self.parse_newline_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
+
+    def parse_structured_field(self, field):
+        return str(field)
+
+    def get_structured_fields(self, ticket):
+        fields = []
+        for field in STRUCTURED_FIELDS:
+            if field in ticket:
+                fields.append(self.parse_structured_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
+
+    def yield_ticket_batches(self):
+        """Generates batches of tickets"""
+
+        for raw_query in self.settings["raw_queries"]:
+
+            first_page = self.zendesk.search(raw_query=raw_query)
+            yield first_page["results"]
+
+            count = first_page["count"]
+            print "{} tickets from ZenDesk match filter.".format(count)
+            page_count, remainder = divmod(count, self.batch_size)
+            page_count = page_count + 1 if remainder else page_count
+
+            for page_id in range(1, page_count):
+                qpage = " page:{}".format(page_id)
+                page = self.zendesk.search(raw_query=raw_query + qpage)
+                yield page["results"]
```

Comment:
Not a biggie, but if you loop over the results here and `yield` each result separately you can abstract the fact that you're doing batches from the caller.